docs/site/migration/models/mixins.md

```diff
@@ -6,8 +6,805 @@ sidebar: lb4_sidebar
+  Note,
+  typeof Note.prototype.id,
+  NoteRelations
+> {
+  constructor(@inject('datasources.db') dataSource: DbDataSource) {
+    super(Note, dataSource);
+  }
+}
+```
+
+### Adjust NoteRepository To Use FindByTitleRepositoryMixin
+
+Import the `FindByTitleRepositoryMixin` mixin class factory function, and adjust
+the declaration of the `NoteRepository` class to extend the class returned from
+the mixin function that takes in the `DefaultCrudRepository` superclass an
+input.
+
+{% include code-caption.html content="src/repositories/note.repository.ts" %}
+
+```ts
+import {FindByTitleRepositoryMixin} from '../mixins/findByTitleRepositoryMixin';
+import {DefaultCrudRepository} from '@loopback/repository';
+import {Note, NoteRelations} from '../models';
+import {DbDataSource} from '../datasources';
+import {inject, Constructor} from '@loopback/core';
+
+export class NoteRepository extends FindByTitleRepositoryMixin<
+  Note,
+  Constructor<
+    DefaultCrudRepository<Note, typeof Note.prototype.id, NoteRelations>
+  >
+>(DefaultCrudRepository) {
+  constructor(@inject('datasources.db') dataSource: DbDataSource) {
+    super(Note, dataSource);
+  }
+}
+```
+
+We have now added the `findByTitle` method to a repository via a mixin class
+factory function.
+
+### Defining A Controller Mixin Class Factory Function
+
+In **src/mixins/findByTitleControllerMixin.ts**, let's define the mixin class
+factory function `FindByTitleControllerMixin` which adds the `findByTitle`
+method to any controller.
+
+{% include code-caption.html content="src/mixins/src/mixins/findByTitleControllerMixin.ts" %}
+
+```ts
+import {Constructor} from '@loopback/context';
+import {Model} from '@loopback/repository';
+import {FindByTitleInterface} from './findByTitleInterface';
+import {param, get, getModelSchemaRef} from '@loopback/rest';
+
+export interface FindByTitleControllerMixinParms {
+  basePath: string;
+  modelClass: Constructor<{}>;
+  modelClassName: string;
+}
+
+export function FindByTitleControllerMixin<
+  M extends Model,
+  T extends Constructor<any>
+>(superClass: T, parms: FindByTitleControllerMixinParms) {
+  class MixedController extends superClass implements FindByTitleInterface<M> {
+    constructor(...args: any[]) {
+      super(...args);
+    }
+
+    @get(parms.basePath + '/findByTitle/{title}', {
+      responses: {
+        '200': {
+          description: `Array of ${parms.modelClassName} model instances`,
+          content: {
+            'application/json': {
+              schema: {
+                type: 'array',
+                items: getModelSchemaRef(parms.modelClass, {
+                  includeRelations: true,
+                }),
+              },
+            },
+          },
+        },
+      },
+    })
+    async findByTitle(@param.path.string('title') title: string): Promise<[M]> {
+      let foundItems = await this.repository.findByTitle(title);
+
+      return foundItems;
```

Comment:
Nitpick: the method body can be simplified as follows:

```ts
class MixedController  /*...*/ {
  
  @get(parms.basePath + '/findByTitle/{title}', {
    // ...
  })
  findByTitle(@param.path.string('title') title: string): Promise<[M]> {
    return this.repository.findByTitle(title);
  }
}
```

This is not only less lines of code, but also slightly more performant, because it creates one less Promise object at runtime.

See https://bajtos.net/2018-AsyncAwait/#/9/6 and [promise rewrapping](https://exploringjs.com/es2016-es2017/ch_async-functions.html#_returned-promises-are-not-wrapped)